src/providers/data_provider_be.c

```diff
@@ -689,8 +689,9 @@ int main(int argc, const char *argv[])
         }
     }
 
-    if (be_domain == NULL) {
-        fprintf(stderr, "\nMissing option, --domain is a mandatory option.\n\n");
+    if (!is_syntactically_correct_domain_name(be_domain)) {
+        fprintf(stderr, "\nMissing or malfromed option, "
```

Comment:
Why fprintf() here instead one of our DEBUG() functions?

Reply:
This is original code. I didn't touch it since it's out of the scope of this PR.